mlflow/models/evaluation/default_evaluator.py

```diff
@@ -369,6 +386,26 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _compute_df_mode_or_mean(df):
+    """
+    Compute mean (for continuous columns) and compute mode (for other columns) for the
+    input dataframe, return a dict, key is column name, value is the corresponding mode or
+    mean value, this function calls `_is_continuous` to determine whether the
+    column is continuous column.
+    """
+    df_cols = df.columns.tolist()
+    continuous_cols = [c for c in df_cols if _is_continuous(df[c])]
+    means = df[continuous_cols].mean().to_dict()
+    if continuous_cols == df_cols:
+        return means
```

Comment:
This case `df.drop(continuous_cols)` results is a empty dataframe which causes `.loc[0]` raises key error.